pkg/controller/installation/validation.go

```diff
@@ -331,6 +331,30 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 				return fmt.Errorf("spec.calicoNetwork.containerIPForwarding is supported only for Calico CNI")
 			}
 		}
+
+		type TuningSpec struct {
+			Sysctl *map[string]string `json:"sysctl,omitempty"`
+			Type   string             `json:"type"`
+		}
+
+		if instance.Spec.CalicoNetwork.SysctlTuning != nil {
+			// CNI tuning plugin
+			pluginData := instance.Spec.CalicoNetwork.SysctlTuning
+
+			// sysctl settings
+			allowedKeys := map[string]struct{}{
+				"net.ipv4.tcp_keepalive_intvl":  {},
+				"net.ipv4.tcp_keepalive_probes": {},
+				"net.ipv4.tcp_keepalive_time":   {},
+			}
+
+			for key, value := range *pluginData {
+				if _, allowed := allowedKeys[key]; !allowed {
+					return fmt.Errorf("value %s not allowed in spec.calicoNetwork.sysctlTuning", value)
+				}
+			}
+
+		}
```

Comment:
This can all be removed since we'll depend on Kubebuilder validation.

Reply:
Let's make the change to use kubebuilder validation, but also leave this here as a second line of defense since kubebuilder validation is circumventable by modifying the CRDs.